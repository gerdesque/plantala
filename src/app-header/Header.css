.app-logo {
  height: 4em;
  width: 4em;
  margin-right: 1em;
}
