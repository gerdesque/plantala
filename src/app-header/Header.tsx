import React from 'react';
import { AppBar, IconButton, Toolbar, Typography } from '@material-ui/core';
import InvertColorsIcon from '@material-ui/icons/InvertColors';
import VolumeOffIcon from '@material-ui/icons/VolumeOff';
import VolumeUpIcon from '@material-ui/icons/VolumeUp';
import { makeStyles } from '@material-ui/core/styles';
import logo from '../assets/logo.webp';

interface IHeaderProps {
  setStart: any
  isPlaying: boolean
  setSound: any
  setColorMode: any
}

const useStyles = makeStyles((theme) => ({
  header: {
    backgroundColor: theme.palette.primary.main
  },
  icon: {
    marginRight: theme.spacing(2),
    height: theme.spacing(8),
    width: theme.spacing(8),
    cursor: 'pointer'
  },
  title: {
    flexGrow: 1,
  },
}));

export default function Header({ setStart, isPlaying, setSound, setColorMode }: IHeaderProps) {
  const classes = useStyles();

  return (
    <div>
      <AppBar position="relative" className={classes.header}>
        <Toolbar>
          <img src={logo} className={classes.icon} alt="Plantala logo" onClick={() => setStart()} />
          <Typography variant="h2" className={classes.title}>
            Plantala
          </Typography>
          <IconButton aria-label="invert" onClick={() => setSound(isPlaying)}>
            {isPlaying ? <VolumeUpIcon /> : <VolumeOffIcon />}
          </IconButton>
          <IconButton aria-label="invert" onClick={() => setColorMode()}>
            <InvertColorsIcon />
          </IconButton>
        </Toolbar>
      </AppBar>
    </div>
  )
}
