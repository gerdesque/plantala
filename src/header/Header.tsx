import React from 'react';
import logo from '../logo.png';
import './Header.css';

const Header = () => 
  <div className="app-header">  
    <img src={logo} className="app-logo" alt="logo" />
    <h1>Plantala</h1>
  </div>

export default Header;
