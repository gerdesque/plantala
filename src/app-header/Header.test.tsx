import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Header from './Header';
import AppBar from '@material-ui/core/AppBar';
import Typography from '@material-ui/core/Typography';

describe('Header', () => {

  let shallow: any;
  let wrapper: any;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<Header />);
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div /> container', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the <AppBar /> Component', () => {
    expect(wrapper.containsMatchingElement(AppBar)).toEqual(true);
  });

  it('should display a logo', () => {
    expect(wrapper.find('img').length).toEqual(1);
  });

  it('should render the <Typography /> Component', () => {
    expect(wrapper.containsMatchingElement(Typography)).toEqual(true);
  });

  it('should render the title "Plantala"', () => {
    expect(wrapper.find(Typography).text()).toEqual('Plantala');
  });
  

});
