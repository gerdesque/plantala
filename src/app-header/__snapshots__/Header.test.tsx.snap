// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div>
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-header-1"
    position="relative"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <img
        alt="Plantala logo"
        className="makeStyles-icon-2"
        onClick={[Function]}
        src="logo.webp"
      />
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        variant="h2"
      >
        Plantala
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="play music"
        onClick={[Function]}
      >
        <VolumeOffIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="invert color"
        onClick={[Function]}
      >
        <InvertColorsIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="about"
        onClick={[Function]}
      >
        <InfoIcon />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
